### DISCUSSION TITLE
Dashboard Url in Reporter

### DISCUSSION THREAD

**Discussion by dracahn** [cypress-io/cypress#26860] [Questions and Help]
Hello,
I'm trying to make a reporter that include the runUrl in the report.  I'm hoping to get a link from the report directly to the run in cypress dashboard.  Is there any way to retrieve that variable for a reporter?  I haven't been able to find it.  Specifically: at the end of a run, there's a console log line that says `Recorded Run: https://cloud.cypress.io/projects/{projectId}/runs/{runNumber}`.  I would like to put that url in my report.

E.G.:
```
class MyCustomReporter {
    constructor(runner, options) {
        runner.once(EVENT_RUN_BEGIN, () => {
            console.log('start');
            // I'd like to log it here.  If I know how to log it here, I can put it where i need it to go.
            console.log(`RunUrl: ${someRunUrl}`); // RunUrl: https://cloud.cypress.io/projects/{projectId}/runs/{runNumber}
        })
    }
}```

**Comment by aevans-mms** [comment]
I'm not sure if you can directly get the Cypress **runUrl** in a custom reporter, and I haven't found a way to get the **runId** directly to recreate the URL with your **projectId**.

But, you can use the [Cypress Node.js Module API](https://docs.cypress.io/guides/guides/module-api) to launch your test run and get the runUrl.  Something like this:  

```
// custom_runner.js

var cypress = require('cypress')
var fs = require('fs')

var options = { 
	record: true, 
	key: process.env.CYPRESS_KEY, 
	reporter: 'custom_reporter.js', 
	reporterOptions: { dataFile: 'runUrl.json'},
	spec: "**/e2e/**"
}

cypress.run(options)
	.then((results)=>{
		console.log("runUrl: ", results.runUrl)
		
		// save the runUrl to a file so we can read it later
		fs.writeFileSync(
			options.reporterOptions.dataFile, 
			JSON.stringify({runUrl: results.runUrl})
		)
	})
	.catch((err) => {
		console.error(err)
	})
```

and then (this is a bad hack) you can read the runUrl.json in your custom reporter to get the data you need:

```
// custom_reporter.js

var mocha = require('mocha')
var fs = require('fs')

function custom_reporter(runner, options) {
	runner.once(mocha.Runner.constants.EVENT_RUN_END, () => {
		var contents = fs.readFileSync(options.reporterOptions.dataFile, {encoding: 'utf8', flag: 'r' })
		var data = JSON.parse(contents)
		console.log("===> ", data.runUrl)
	})
}

module.exports = custom_reporter
```

It would be ideal if the runUrl (or runId) were attached the the **runner** object

**Reply by aevans-mms** [reply]
It looks like this may not work either because `cypress.run(options).then()` is only triggered after the EVENT_RUN_END.

So you would need to append the runUrl after writing the report!
